cmd/kube-proxy/app/server.go

```diff
@@ -778,6 +784,54 @@ func (s *ProxyServer) birthCry() {
 	s.Recorder.Eventf(s.NodeRef, api.EventTypeNormal, "Starting", "Starting kube-proxy.")
 }
 
+func (s *ProxyServer) reportVersion() error {
+	v := version.Get().String()
+	nodeClient := s.CoreClient.Nodes()
+	nodeName := s.NodeRef.Name
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		oldNode, err := nodeClient.Get(nodeName, metav1.GetOptions{})
+		if err != nil {
+			glog.Warningf("Failed to retrieve node info: %v", err)
+			return err
+		}
+		clonedNode, err := api.Scheme.DeepCopy(oldNode)
```

Comment:
Do you need to do a deep copy, since you're using a client to retrieve oldNode? Could you get oldNode, marshal to oldData, set the version, marshal to newData, and go from there?